Source/Core/Core/Boot/AncastTypes.h

```diff
@@ -0,0 +1,85 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "Common/CommonTypes.h"
+
+// Magic number
+#define ANCAST_MAGIC 0xEFA282D9
+
+// The location in memory where PPC ancast images are booted from
+#define ESPRESSO_ANCAST_LOCATION_PHYS 0x01330000
+#define ESPRESSO_ANCAST_LOCATION_VIRT 0x81330000
+
+// The image type
+enum AncastImageType
+{
+    ANCAST_IMAGE_TYPE_ESPRESSO_WIIU = 0x11,
+    ANCAST_IMAGE_TYPE_ESPRESSO_WII  = 0x13,
+    ANCAST_IMAGE_TYPE_STARBUCK_NAND = 0x21,
+    ANCAST_IMAGE_TYPE_STARBUCK_SD   = 0x22,
+};
+
+// The console type of the image
+enum AncastConsoleType
+{
+    ANCAST_CONSOLE_TYPE_DEV     = 0x01,
+    ANCAST_CONSOLE_TYPE_RETAIL  = 0x02,
+};
+
+// Start of each header
+struct AncastHeaderBlock
+{
+    u32 magic;
+    u32 unknown;
+    u32 header_block_size;
+    u8 padding[0x14];
+};
+static_assert(sizeof(AncastHeaderBlock) == 0x20, "AncastHeaderBlock has the wrong size");
```

Comment:
```suggestion
static_assert(sizeof(AncastHeaderBlock) == 0x20);
```
The assertion message for an static_assert is optional (since C++17). In this case it doesn't really add much so I think we should get rid of it